src/services/tasks/types.ts

```diff
@@ -41,6 +41,15 @@ export type TaskInput = Pick<
   'functionName' | 'data' | 'startTimestamp' | 'priority' | 'tags'
 >;
 
+export interface DirectExecutionResponse<T, U> extends Task<U> {
+  /** The Extra Horizon document id for the application used to make the request */
+  createdByApplicationId?: ObjectId;
+  /** The Extra Horizon document id for the user who made the request */
+  createdByUserId?: ObjectId;
```

Comment:
(minor) I belief these fields should actually be part of the Task interface itself